do_cmake.sh

```diff
@@ -42,7 +42,7 @@ if [ -r /etc/os-release ]; then
       ubuntu)
           MAJOR_VER=$(echo "$VERSION_ID" | sed -e 's/\..*$//')
           if [ "$MAJOR_VER" -ge "22" ] ; then
-              PYBUILD="3.10"
+              PYBUILD="3.12"
```

Comment:
why do need this change?

Reply:
I needed this on my ubuntu 24.04. 
I could have installed python-3.10 as an alternative.  But it was not available in the package repo anymore.

But we can remove this without issue till fedora 40

Reply:
yes, this will probably break lots of stuff.
you can try and run `cmake` with `-DWITH_PYTHON3=3.12` ?